spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java

```diff
@@ -1239,6 +1239,108 @@ public void testWriteSchemaSubset() throws Exception {
     assertThat(testTableNumberOfRows()).isEqualTo(3);
   }
 
+  @Test
+  public void allowFieldAdditionWithNestedColumns() throws Exception {
+    assumeThat(writeMethod, equalTo(WriteMethod.INDIRECT));
+    StructType initialSchema =
+        structType(
+            StructField.apply("value", DataTypes.StringType, true, Metadata.empty()),
+            StructField.apply("ds", DataTypes.DateType, true, Metadata.empty()));
+    List<Row> rows =
+        Arrays.asList(
+            RowFactory.create("val1", Date.valueOf("2023-04-13")),
+            RowFactory.create("val2", Date.valueOf("2023-04-14")));
+    Dataset<Row> initialDF = spark.createDataFrame(rows, initialSchema);
+    // initial write
+    initialDF
+        .write()
+        .format("bigquery")
+        .mode(SaveMode.Overwrite)
+        .option("dataset", testDataset.toString())
+        .option("table", testTable)
+        .option("temporaryGcsBucket", TestConstants.TEMPORARY_GCS_BUCKET)
+        .option("intermediateFormat", "avro")
+        .option("writeMethod", writeMethod.toString())
+        .save();
+    assertThat(testTableNumberOfRows()).isEqualTo(2);
+
+    StructType finalSchema =
+        structType(
+            StructField.apply("value", DataTypes.StringType, true, Metadata.empty()),
+            StructField.apply("ds", DataTypes.DateType, true, Metadata.empty()),
+            StructField.apply("bool_type", DataTypes.BooleanType, true, Metadata.empty()));
+    List<Row> finalRows =
+        Arrays.asList(
+            RowFactory.create("val3", Date.valueOf("2023-04-15"), true),
+            RowFactory.create("val4", Date.valueOf("2023-04-16"), false));
+    Dataset<Row> finalDF = spark.createDataFrame(finalRows, finalSchema);
+    finalDF
+        .write()
+        .format("bigquery")
+        .mode(SaveMode.Append)
+        .option("dataset", testDataset.toString())
+        .option("table", testTable)
+        .option("temporaryGcsBucket", TestConstants.TEMPORARY_GCS_BUCKET)
+        .option("writeMethod", writeMethod.toString())
+        .option("allowFieldAddition", "true")
+        .option("allowFieldRelaxation", "true")
+        .save();
+    assertThat(testTableNumberOfRows()).isEqualTo(4);
+
+    StructType schema3 =
+        structType(
```

Comment:
Please move case3 into a separate test (each test should check only one thing) and add case 4 from [here](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/issues/945#issuecomment-1580422730)

Reply:
Addressed